kernel/kheap.c

```diff
@@ -51,6 +51,31 @@ void *k_heap_alloc(struct k_heap *h, size_t bytes, k_timeout_t timeout)
 	return ret;
 }
 
+void *k_heap_aligned_alloc(struct k_heap *h, size_t align, size_t bytes, k_timeout_t timeout)
```

Comment:
Rather than cutting and pasting the synchronization as boilerplate from k_heap_alloc(), could you please refactor the two so they use the same implementation?  One simple option would be to implement k_heap_alloc() in terms of k_heap_aligned_alloc().  Or if we care about the mild performance overhead (IMHO we don't) implement both in terms of an inline function where the alignment code optimizes away.